### DISCUSSION TITLE
Gitlab platformAutomerge is not working when pipeline creation takes more than 7.5 sec.

### DISCUSSION THREAD

**Discussion by fullstackcreatives** [renovatebot/renovate#25946] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

Platform Automerge in Gitlab will set auto-merge in Gitlab when Merge Request option is set to "pipelines must succeed". Automerge can only be activated when the merge request has a pipeline associated to it. In some Gitlab installations the creation of a pipeline can take some time.
Renovate tries to set auto-merge here  https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/gitlab/index.ts#L650 latest after 7.5 sec timeout which on our platform is most of the times not enough time to create a pipeline. There for platform automerge is not set in the merge requests.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Which server/service is setting the pipeline and why does it take more than 7.5s? e.g. is this an internal gitlab thing or some external CI like Jenkins?

**Reply by fullstackcreatives** [reply]
It is an internal Gitlab (no external CI). Gitlab Runners are hosted on self-hosted Openshift/EKS or as an alternative in AWS EC2.

BTW: I have added a PR to solve this: https://github.com/renovatebot/renovate/pull/25947

**Reply by rarkins** [reply]
What would be the downside if we increased the retry intervals to something high like 5 minutes to avoid the need for another setting? I think it might be a better trade-off. If it takes too long then the user can always disable the platform automerge setting.

**Reply by fullstackcreatives** [reply]
From my side would be ok. I assumed that you would like to have an option for this. I would suggest to increase the number of retries so that at the end it runs a maximum of 5 minutes. But please add the memCache: false option to the request like I did in my PR here: https://github.com/renovatebot/renovate/pull/25947/files#diff-bd7375b4be9976d8d1ee7d683faf833d11b283e65518c59be09ace09271b90e8R656
Otherwise the function will always run 5 minutes (see https://github.com/renovatebot/renovate/discussions/25945)

I could edit my PR accordingly if that would help to have it in production as soon as possible?!

**Reply by rarkins** [reply]
Yes, the memCache=false option is a fix, can be separate. Ideally we prefer bugs fixed separately so that the chances of the PR needing reverting is lower

**Reply by fullstackcreatives** [reply]
Here is the PR for memCache fix: https://github.com/renovatebot/renovate/pull/25949

**Reply by fullstackcreatives** [reply]
As discussed here I increased the retryTimes to 35 which results in a total timeout of appr. 5 minutes. I closed the original PR and created a new one: https://github.com/renovatebot/renovate/pull/25953
